CRM/Contact/Page/AJAX.php

```diff
@@ -325,17 +324,19 @@ public static function deleteCustomValue() {
    */
   static public function checkUserName() {
     $signer = new CRM_Utils_Signer(CRM_Core_Key::privateKey(), array('for', 'ts'));
+    $sig = CRM_Utils_Request::retrieve('sig', 'String', CRM_Core_DAO::$_nullObject);
+    $for = CRM_Utils_Request::retrieve('for', 'String', CRM_Core_DAO::$_nullObject);
     if (
       CRM_Utils_Time::getTimeRaw() > $_REQUEST['ts'] + self::CHECK_USERNAME_TTL
-      || $_REQUEST['for'] != 'civicrm/ajax/cmsuser'
-      || !$signer->validate($_REQUEST['sig'], $_REQUEST)
+      || $for != 'civicrm/ajax/cmsuser'
+      || !$signer->validate($sig, $_REQUEST)
     ) {
       $user = array('name' => 'error');
       CRM_Utils_JSON::output($user);
     }
 
     $config = CRM_Core_Config::singleton();
-    $username = trim($_REQUEST['cms_name']);
+    $username = trim(CRM_Utils_Type::escape($_REQUEST['cms_name'], 'String'));
```

Comment:
Actually we are hitting the DB on line 344: $config->userSystem->checkUserNameEmailExists($params, $errors);
But these function already do $dao->escape(), so we might be double escaping if the username has any quotes. Also, the list of characters you are suggesting might be adequate for all CMS's, so we should rely on the CRM_Utils_System_XXX to do the work.

So ... would a simple CRM_Utils_Array::value('cms_name', $_REQUEST) be sufficient here AND quiet the Fortify scanner?